src/main/java/uk/ac/cam/cl/dtg/segue/api/AuthorisationFacade.java

```diff
@@ -104,20 +104,54 @@ public AuthorisationFacade(final PropertiesLoader properties, final UserAccountM
     }
 
     /**
-     * Get all users who can see my data.
-     * 
+     * Get all users who can see the current user's data.
+     *
      * @param request
      *            - so we can identify the current user.
+     * @throws NoUserException
+     *            - when the target user cannot be found.
      * @return List of user associations.
      */
     @GET
     @Path("/")
     @Produces(MediaType.APPLICATION_JSON)
     @GZIP
     @ApiOperation(value = "List all users granted access to the current user's data.")
-    public Response getUsersWithAccess(@Context final HttpServletRequest request) {
+    public Response getUsersWithAccess(@Context final HttpServletRequest request) throws NoUserException {
         try {
-            RegisteredUserDTO user = userManager.getCurrentRegisteredUser(request);
+            RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+            return getUsersWithAccessSpecificUser(request, requestingUser.getId());
+        } catch (NoUserLoggedInException e) {
+            return SegueErrorResponse.getNotLoggedInResponse();
+        }
+    }
+
+    /**
+     * Get all users who can see the specified user's data.
+     * 
+     * @param request
+     *            - so we can identify the current user.
+     * @param userId
+     *            - the user id we want the associations of.
+     * @throws NoUserException
+     *            - when the target user cannot be found.
+     * @return List of user associations.
+     */
+    @GET
+    @Path("/{userId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "List all users granted access to the specified user's data.")
+    public Response getUsersWithAccessSpecificUser(@Context final HttpServletRequest request, @PathParam("userId") Long userId) throws NoUserException {
+        try {
+            RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+
+            if (!isUserStaff(userManager, request) && !userId.equals(requestingUser.getId())) {
```

Comment:
It's a bit late now, but this was missed by whoever reviewed this PR. On the most common path (viewing users who have access to the current user's data), we load the user object from the database 3 times. Once to get the ID in the first method, once to get `requestingUser` and then again to do the `isUserStaff` check. This is a bit much!

We could avoid loading the user more than once by passing `null` rather than a user ID, but that seems a bit hideous. I think we live with loading the user twice, and use `isUserStaff(userManager, requestingUser)` instead!

Reply:
Oh, it's actually four times, because we then do `getUserDTOById(userId)` below, too!

Reply:
See https://github.com/isaacphysics/isaac-api/commit/b0082b47213e2a77f147c2cb3d856bde136c7648 for fix.